debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisStreamChangeConsumer.java

```diff
@@ -118,62 +103,106 @@ void close() {
         catch (Exception e) {
             LOGGER.warn("Exception while closing Jedis: {}", client, e);
         }
+        finally {
+            client = null;
+        }
+    }
+
+    /**
+    * Split collection to batches by batch size using a stream
+    */
+    private <T> Stream<List<T>> batches(List<T> source, int length) {
+        if (length <= 0) {
+            throw new IllegalArgumentException("length = " + length);
+        }
```

Comment:
How about we validate validity of the configured batch size once, in the post construct method?

Reply:
Yes, that's better.

Reply:
Although, this actually reveals that we don't validate the properties, for example - initialRetryDelay must be < than maxRetryDelay. Do you have such a mechanism?

Reply:
Yes, good idea, looks like this deserves more general handling indeed. Could you log an issue in Jira for this?